### ISSUE TITLE
Need a mechanism to map Amazon API generated Boxes with sofotware generated boxes

### ISSUE THREAD

**Issue by midi-red** [amzn/selling-partner-api-models#4473]
When calling [setPackingInformation](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#setpackinginformation), we pass box content information. We need to pass some metadata for each pack so that we can re-map API generated boxes with our system generated boxes. 
Amazon generated box ids like FBA1234567U000001 - FBA1234567U00000n, then for another shipment in the same shipmen plan FBAABCDEFU000001 - FBAABCDEFU00000m.
Although FBA1234567U000001 will map to say, B1 in a third party software, FBAABCDEFU000001 will map to, say B25. We try to match the boxes by matching the box contents, but it is very error prone when putting shipment labels on the physical boxes. 

If you accept a metadata field in [BoxInput](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#boxinput) and return it back when we call listShipmentBoxes, that will solve the issue.

**Comment by pretherford** [comment]
Agreed this will help but the shipping labels do show the mapping of pack group and box # to shipment box #. In the pack first flow, each box should be labeled with an identifier, e.g. P1 - B25. Take a look at the Amazon generated box labels. They show the pack group and box # underneath and are printed in shipment order. The box label prints 1st followed by the shipping label.

**Comment by puppsupr** [comment]
Hi @midi-red,
While we improve this experience, the current workaround is to use [listInboundPlanBoxes](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#listinboundplanboxes), then match the templateName with [getLabels](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v0-reference#getlabels) and box IDs.

**Comment by midi-red** [comment]
This is especially critical for 2D Barcode content information source. There is no way to match boxes since you do not provide box content when calling setPackingInformation. We really need a meta field in BoxInput when calling setPackingInformation so that we can match with actual box.

**Comment by MeenaAmz** [comment]
Hello @midi-red and @pretherford,

Thank you for proposing this enhancement. We have submitted the feature request for review. 

Thanks,
Meena
Selling Partner API Developer Services
